File: frontend/src/types/shop.ts
// 가게 등록 요청 타입
export interface ShopRequest {
  name: string
  address: string
  location: { lat: number; lng: number }
  categoryIds: string[]
  phone: string
  businessHours: string
}

// 메뉴 항목 타입
export interface MenuItem {
  name: string
  price: number
}

// 가게에 포함된 카테고리 정보
export interface CategoryInfo {
  id: string
  name: string
}

// 가게 타입 정의
export interface Shop {
  id: string
  name: string
  address: string
  location: {
    lat: number
    lng: number
  }
  categories: CategoryInfo[]
  phone?: string
  businessHours?: string
  menu: MenuItem[]
  reviewCount: number
  averageRating: number
  imageUrl?: string
}

File: backend/src/main/kotlin/com/ramap/model/Shop.kt
package com.ramap.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

// 가게 위치 좌표
data class GeoLocation(
    val lat: Double,
    val lng: Double
)

// 가게 메뉴 항목
data class MenuItem(
    val name: String,
    val price: Int
)

// 라멘 가게 MongoDB 도큐먼트
@Document(collection = "shops")
data class Shop(
    @Id
    val id: String? = null,
    val name: String,
    val address: String,
    val location: GeoLocation,
    val categoryIds: List<String> = emptyList(),
    val phone: String = "",
    val businessHours: String = "",
    val menu: List<MenuItem> = emptyList(),
    val averageRating: Double = 0.0,
    val reviewCount: Int = 0,
    val status: String = "pending", // pending | approved | rejected
    val createdAt: LocalDateTime = LocalDateTime.now()
)
